### ISSUE TITLE
Support bulk release file delete via API

### ISSUE THREAD

**Issue by nefrob** [getsentry/sentry#50613]
### Problem Statement

TLDR: add a bulk release file delete endpoint for use in code (ex. cron cleanup job on releases inactive for 30 days).

For releases with health data that cannot be deleted via `/api/0/organizations/{organization_slug}/releases/{version}/` it would be nice to be able to clean up all the release artifacts (ex. source maps) in one bulk operation. Right now there is a single file delete endpoint `/api/0/projects/{organization_slug}/{project_slug}/releases/{version}/files/{file_id}/` however if a release has hundreds or thousands of files and releases are created many times a day this is not a great option.

Other investigation shows the `sentry-cli` allows for bulk file delete, however it does not mesh well with a programmatic solution that does other filtering on release data (like the cron example above).

### Solution Brainstorm

Add an endpoint without file detail that deletes all of a release's files:

DELETE
/api/0/projects/{organization_slug}/{project_slug}/releases/{version}/files/

### Product Area

Releases
